.github/workflows/run-tests.yml

```diff
@@ -109,35 +108,32 @@ jobs:
           name: lcov.info
           path: coverage/lcov.info
 
-      - name: Get Sonar coverage
-        id: get-sonar-coverage
+      - name: Get active branch coverage
+        id: get-active-branch-coverage
+        run: |
+          active_branch_coverage=$(yarn nyc report --reporter=text-summary | grep 'Lines' | awk '{gsub(/%/, ""); print int($3)}')
+          echo "The active branch coverage is $active_branch_coverage%."
+          echo 'ACTIVE_BRANCH_COVERAGE='"$active_branch_coverage" >> "$GITHUB_OUTPUT"
+
+      - name: Get default branch coverage
+        id: get-default-branch-coverage
         env:
-          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPOSITORY_OWNER: ${{ github.repository_owner }}
+          REPOSITORY: ${{ github.repository }}
         run: |
-          projectKey=$(grep 'sonar.projectKey=' sonar-project.properties | cut -d'=' -f2)
-          sonar_coverage=$(curl --silent --header "Authorization: Bearer $SONAR_TOKEN" "https://sonarcloud.io/api/measures/component?component=$projectKey&metricKeys=coverage" | jq -r '.component.measures[0].value // 0')
-          echo "The Sonar coverage of $projectKey is $sonar_coverage%."
-          echo 'SONAR_COVERAGE='"$sonar_coverage" >> "$GITHUB_OUTPUT"
+          default_branch_coverage=$(gh api -H "Accept: application/vnd.github.raw" "repos/$REPOSITORY_OWNER/$REPOSITORY/contents/coverage.json" | jq -r ".coverage // 0")
```

Comment:
We can get previous test coverage `coverage.json` file. Why query for it dynamically? This only makes the quality gate more confusing to understand, because it's unclear which version of the main branch it's comparing against.

The merge queue will ensure the quality gate is checked with an up-to-date version of `coverage.json`, so we can use the local file instead. This will have more consistent results, easier to understand and replicate.

This approach would also be problematic for PRs targeting branches other than `develop`. Release related PRs for example, or PRs against another person's feature. We don't want PRs like that blocked due to changes for reasons that have nothing to do with the PR contents.